templates/sqoop-parquet-hdfs-hive-merge/create-merge-view.sql

```diff
@@ -0,0 +1,30 @@
+{#  Copyright 2017 Cargill Incorporated
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License. #}
+
+set sync_ddl=1;
+USE {{ conf.staging_database.name }};
+
+--Create merged view --
+DROP VIEW IF EXISTS {{ table.destination.name }}_view;
+CREATE VIEW {{ table.destination.name }}_view AS
+  SELECT t1.* 
+  FROM (SELECT * FROM {{ table.destination.name }}_base
+	UNION ALL 
+	SELECT * FROM {{ table.destination.name }}_incr) t1
+  JOIN (SELECT {{ table.split_by_column }}, max({{ table.check_column }}) max_modified 
```

Comment:
Is the assumption here that the `split_by_column` is the primary key (because this is where the primary key is needed). I think this is confusing. You could instead add a `primary_key` field to the `tables.yml` file.
For an advanced improvement, you could support multiple primary keys (composite keys). Though being able to do this merge on just a single primary key is useful

Reply:
Yes. And I used split_by_column simply because I failed to notice that the primary_keys are already defined in tables.yml. Switched to using primary_keys instead.